implement missing onPause and onResume handlers on certain transformers

frankpepermans opened the pull request:
see https://github.com/ReactiveX/rxdart/issues/156

brianegan commented on lib/src/streams/race.dart:
```diff
@@ -66,6 +66,14 @@ class RaceStream<T> extends Stream<T> {
           }
         },
+        onPause: ([Future<dynamic> resumeSignal]) => subscriptions
+            .where((StreamSubscription<T> subscription) => subscription != null)
```
Is the null check necessary here?

frankpepermans commented:
I've added the ```pause```/```resume``` tests, plus I cleaned up ```amb```/```race``` a bit.
We had a fixed-length ```Subscription``` ```List``` before, once the race was over, all obsolete ```Subscription```s in that ```List``` were cancelled and then their reference in that List was set to ```Null```.

...Which is why there was a ```Null``` check in place when iterating over that ```List```...

Now I simply remove the obsolete ```Subscription```s from the ```List```, so only the active ```Stream``` remains in there.
